### PR TITLE
Prolongs dfg graphs to the root of a language feature 

### PR DESCRIPTION
**Pull request by konradweiss**
Currently we have the root node as branching node to keep the semantic information of the language-feature that branches while traversing the EOG. To keep EOG and DFG coherent and later build a propper PDG we prolong the dfg paths to the branching node.

I added the DFG edges where I added the EOG-Edges to the root node: in the EOG-pass- A strategy pattern would be nice to make the root node configurable for EOG and DFG but so far I could not find an easy way to integrate one for the EOG pass.

Root node of language feature if is a sink for the EOG of the condition and then branches out. For certain applications this rootnode allows traversals to use the root node like a jumpi 
eog_root_node

DFG edge from condition to the language-feature root node that acts as branching condition.
dfg_root_node

### CODE REVIEW

**Comment by konradweiss** [comment]
I also tested this with codyze and the tests run through so it should be fine

File: src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.java [hunk 1]
```diff
@@ -781,7 +782,10 @@ private void handleDoStatement(@NonNull Node node) {
     createEOG(doStatement.getStatement());
 
     createEOG(doStatement.getCondition());
+
+    doStatement.addPrevDFG(doStatement.getCondition());
```
Comment (oxisto) [inline src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.java:786]: I am semantically confused about an evaluation order pass, that adds data flow edges. Does that make sense?

**Review by vfsrfs — APPROVED** [review]

**Comment by oxisto** [comment]
> Currently we have the root node as branching node to keep the semantic information of the language-feature that branches while traversing the EOG. To keep EOG and DFG coherent and later build a propper PDG we prolong the dfg paths to the branching node.
> 
> 
> 
> I added the DFG edges where I added the EOG-Edges to the root node: in the EOG-pass- A strategy pattern would be nice to make the root node configurable for EOG and DFG but so far I could not find an easy way to integrate one for the EOG pass.
> 
> 
> 
> Root node of language feature if is a sink for the EOG of the condition and then branches out. For certain applications this rootnode allows traversals to use the root node like a jumpi 
> 
> eog_root_node
> 
> 
> 
> DFG edge from condition to the language-feature root node that acts as branching condition.
> 
> dfg_root_node
> 
> 
> 
> 

Could you provide the code associated with the example graph? I am still not quite sure if I understand it correctly.

**Comment by konradweiss** [comment]
> > Currently we have the root node as branching node to keep the semantic information of the language-feature that branches while traversing the EOG. To keep EOG and DFG coherent and later build a propper PDG we prolong the dfg paths to the branching node.
> > I added the DFG edges where I added the EOG-Edges to the root node: in the EOG-pass- A strategy pattern would be nice to make the root node configurable for EOG and DFG but so far I could not find an easy way to integrate one for the EOG pass.
> > Root node of language feature if is a sink for the EOG of the condition and then branches out. For certain applications this rootnode allows traversals to use the root node like a jumpi
> > eog_root_node
> > DFG edge from condition to the language-feature root node that acts as branching condition.
> > dfg_root_node
> 
> Could you provide the code associated with the example graph? I am still not quite sure if I understand it correctly.

`
if(num > maxSoFar){
...
}
`

**Pull request merged**